### ISSUE TITLE
Pasting issues in WezTerm with tmux on macOS Sequoia when using extended key settings

### ISSUE THREAD

**Issue by john-persson-polestar** [wezterm/wezterm#6291]
### What Operating System(s) are you seeing this problem on?

macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

Yabai

### WezTerm version

I get: "wezterm not found" when running this command

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I initially set out to configure tmux and WezTerm to accept the key combination "Ctrl+Enter". To achieve this, I added the following settings to my tmux config:

```sh
set -g default-terminal "xterm-256color"
set -g extended-keys on
set -as terminal-features 'xterm*:extkeys'
```

While these settings successfully enabled Ctrl+Enter functionality, they introduced an unexpected side effect: pasting text now results in abnormal behavior. When pasting, escape sequences are inserted into the content, particularly replacing return characters. For example, pasting a simple enum from the clipboard results in this output:
```sh
const enum PageComponent {[27;5;106~  FORM = "form",[27;5;106~  SUCCESS = "success",[27;5;106~  ERROR = "error",[27;5;106~}[27;5;106~
```
This issue occurs only in WezTerm when used with tmux and the extended key settings. The same tmux configuration works correctly in iTerm2, suggesting the problem is specific to WezTerm's interaction with tmux when these extended key settings are enabled.

I'm able to reproduce this issue with the default WezTerm configuration (i..e no configuration of my own), and have also tried multiple options but failed to find anything that works.

### To Reproduce

1. Start WezTerm
2. Launch tmux using the configuration provided above
3. Copy a multi-line text snippet (e.g., code block) to your clipboard
4. Paste the text into the tmux session in WezTerm (using Cmd+V or right-click paste)
5. Observe that the pasted text includes unexpected escape sequences, particularly replacing return characters

### Configuration

Default configuration, i.e. nothing custom.

### Expected Behavior

Pasting without escape sequence with the following tmux configuration :
```sh
set -g default-terminal "xterm-256color"
set -g extended-keys on
set -as terminal-features 'xterm*:extkeys'
```

### Anything else?

I'm on the latest nightly build.

Running
```
wezterm -V
```
Gives me the message:
```
wezterm not found
```

**Comment by tlvenn** [comment]
I can confirm the same thing happen on Sonoma and with NixOs

**Comment by abhinavnatarajan** [comment]
I can confirm the same behaviour on PopOS 22.04 with Wezterm 20240203-110809-5046fc22 and tmux 3.5a. However, the strange thing is that these escape sequences are only produced on neovim 0.10.1, and not in other programs (pasting in vim proceeds without issue). The issue disappears when:
1. I use Gnome terminal, or;
2. I don't use tmux, or;
3. I paste into vim instead of neovim, or;
4. I have `set -s extended-keys off` in my `tmux.conf`. It does not matter whether I have `set -as ',xterm*:extkeys'` in `tmux.conf` or not.

**Comment by john-persson-polestar** [comment]
@abhinavnatarajan , intersting observation. I thought for sure I was getting the same behaviour when pasting in my shell (zsh), but now I'm only able to reproduce it in NeoVim. I'm on version `0.10.1` as well.

**Comment by chenxin-yan** [comment]
I am experiencing the same issue. When I paste from the clipboard outside of tmux using `cmd+v` in WezTerm, it works fine, but it does not work inside a tmux session.

**Comment by JohnnyZig** [comment]
I am experiencing same issue on FreeBSD when pasting in neovim v0.10.2 run inside tmux inside wezterm.
Pasting same text in neovim run inside tmux inside xterm works. Also works with wezterm and neovim without tmux.

**Comment by john-persson-polestar** [comment]
@JohnnyZig yes, but the reason I'm here s because I want to have `extended-keys` set to `on` 😄
Setting it to `off` is not a fix to this issue.

**Comment by chenxin-yan** [comment]
I am currently using a workaround that sets a keymap to execute a terminal command, which pastes from the clipboard into tmux. If you don't mind, press two keys instead of using native `cmd+v` to paste. I think it should work.

In `.tmux.conf`, add this:
`bind-key p run "pbpaste | tmux load-buffer -; tmux paste-buffer"`

**Comment by liujoey** [comment]
Same on MacOS, `extended-keys` is `on` in tmux, must be in neovim, must use Cmd-V to paste. (If use vim p key to paste, then everything works as expected). Something to do with system clipboard?

**Comment by john-persson-polestar** [comment]
My workaround has been to left click with the mouse and click "Paste". Not break but I do it rarely enough that it's been fine for now.

Pasting in this way works as expected without escape sequences being added, and to me indicates that its' not related to the system clipboard @liujoey .

**Comment by nikklassen** [comment]
I followed the suggestion and I built tmux from HEAD. It appears that this fixed the problem for me too with no other changes

**Comment by TimWSpence** [comment]
> I followed the suggestion and I built tmux from HEAD. It appears that this fixed the problem for me too with no other changes

Yep same here

**Comment by bew** [comment]
It's not a Wezterm issue then, let's close this, have a good day everyone 👋

**Comment by john-persson-polestar** [comment]
Building tmux from HEAD worked for me as well. Such a relive to have this working again.
Cheers for pitching in everyone! 💯
